### DISCUSSION TITLE
main 브랜치 프로텍션 관련

### DISCUSSION THREAD

**Discussion by otterp012** [over-ui/unstyled#29] [General]
현재 저희의 브랜치 구조는 다음과 같습니다.

dev -- 진행 사항을 push하고, 제대로 동작되는지 확인합니다.
main -- dev 브랜치로부터 진행사항을 이어 받아, `lerna version`을 통해 버전을 올리고, `publish` 합니다.

## 문제되는 상황
- main 브랜치에 프로텍션을 걸어놓은 상태라, `lerna version`이 제대로 적용되지 않아 action에 오류가 나고 있습니다.

## 제안 사항
- 실제로, 개발 중 사용하는 브랜치는 `dev` 브랜치며 `dev` 브랜치에서 서로간에 `pr review`를 진행하고 있습니다. 
- 이러한 점에 따라, `dev` 브랜치에 protection을 걸고, `main` 브랜치는 배포자동화를 원활히 사용하고자 프로텍션을 해제하는 방향을 제안드립니다.
- 또 이렇게 사용한다면, `main` 브랜치를 사용하는 이름에 걸맞게 `publish` 브랜치로, 바꾸는 방향은 어떤지 궁금합니다.

**Comment by lv0314** [comment]
어차피 push는 저희 둘 밖에 하지 못하고, 실직적인 개발은 dev 브랜치에서 이루어지기 때문에 프로텍션을 변경해도 상관 없을 것 같습니다
깃 네이밍 변경은.. 제가 알기로도 그렇고 검색을 해 봐도 대부분 main이나 master를 사용하는 것 같은데 그대로 유지하는게 낫지 않나 싶습니다

**Comment by otterp012** [comment]
프로텍션을, 브랜치 이름에 따라 다르게 걸 수 있어서 `dev` 브랜치로 프로텍션을 걸고 `main`을 디폴트로 유지했습니다!
